crates/proto/src/op/header.rs

```diff
@@ -112,12 +112,18 @@ impl fmt::Display for MessageType {
 /// All the flags of the request/response header
 #[derive(Clone, Copy, PartialEq, Eq, Hash)]
 pub struct Flags {
-    authoritative: bool,
-    truncation: bool,
-    recursion_desired: bool,
-    recursion_available: bool,
-    authentic_data: bool,
-    checking_disabled: bool,
+    /// Authoritative Answer (AA)
+    pub authoritative: bool,
```

Comment:
Why did you make these public? Is that necessary for adding metrics?

Reply:
The fields are now used for request/response metrics.

The `Header::flags()` was already returning the `Flags` struct through the public api, but only `fmt::Display` was used during logging.
As the `Flags` are not likely to change, I've chosen to add the `pub` on the fields in order to be able to use the information during request/response metric creation.

alternatives:
- add `pub` methods on `Flags` to return the boolean values instead of field access
- remove the request / response flag information from the metrics
- use the `Debug` string and parse it

Reply:
Oh, I had missed that the methods are already public available on the Header... :disappointed: 
Updated to use these.